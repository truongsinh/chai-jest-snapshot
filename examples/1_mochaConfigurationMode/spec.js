const expect = require("chai").expect;

describe("using chaiJestSnapshot in mocha-specific configuration mode", function() {
  it("does not require any arguments", function() {
    expect({ foo: "bar" }).to.matchSnapshot();
  });

  it("appends a number to the end of each snapshot name so that if you call matchSnapshot more than once, there isn't a conflict", function() {
    expect({ first: "thing" }).to.matchSnapshot();
    expect({ second: "thing" }).to.matchSnapshot();
  });

  it("uses an optional first argument to update a snapshot", function() {
    expect({ foo: "not bar" }).to.matchSnapshot(true);
  });

  it("can use an environment variable to update the snapshot instead", function() {
    process.env.CHAI_JEST_SNAPSHOT_UPDATE_ALL = "true";
    expect({ foo: "something else" }).to.matchSnapshot();
  });
});
