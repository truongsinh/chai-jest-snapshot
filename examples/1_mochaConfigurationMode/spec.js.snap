exports[`using chaiJestSnapshot in mocha-specific configuration mode appends a number to the end of each snapshot name so that if you call matchSnapshot more than once, there isn't a conflict 1`] = `
Object {
  "first": "thing"
}
`;

exports[`using chaiJestSnapshot in mocha-specific configuration mode appends a number to the end of each snapshot name so that if you call matchSnapshot more than once, there isn't a conflict 2`] = `
Object {
  "second": "thing"
}
`;

exports[`using chaiJestSnapshot in mocha-specific configuration mode can use an environment variable to update the snapshot instead 1`] = `
Object {
  "foo": "something else"
}
`;

exports[`using chaiJestSnapshot in mocha-specific configuration mode does not require any arguments 1`] = `
Object {
  "foo": "bar"
}
`;

exports[`using chaiJestSnapshot in mocha-specific configuration mode uses an optional first argument to update a snapshot 1`] = `
Object {
  "foo": "not bar"
}
`;
